contracts/protocol/pool/PoolInstantWithdraw.sol

```diff
@@ -0,0 +1,571 @@
+    function getLoanDebtValue(uint256 loanId) external view returns (uint256) {
+        DataTypes.PoolStorage storage ps = poolStorage();
+        DataTypes.TermLoanData storage loan = ps._termLoans[loanId];
+        require(
+            loan.state == DataTypes.LoanState.Active,
+            Errors.INVALID_LOAN_STATE
+        );
+
+        uint256 loanDebt = _calculateLoanStableDebt(
+            loan.borrowAmount,
+            loan.discountRate,
+            loan.startTime
+        );
+
+        return loanDebt;
+    }
+
+    /// @inheritdoc IPoolInstantWithdraw
+    function createLoan(
+        address collateralAsset,
+        uint256 collateralTokenId,
+        uint256 collateralAmount,
+        address borrowAsset,
+        uint16 referralCode
+    ) external nonReentrant returns (uint256) {
+        DataTypes.PoolStorage storage ps = poolStorage();
+        _validateBorrowAsset(ps, collateralAsset, borrowAsset);
+
+        DataTypes.ReserveCache memory reserveCache;
+        DataTypes.ReserveData storage reserve = ps._reserves[borrowAsset];
+        reserveCache = reserve.cache();
+        reserve.updateState(reserveCache);
+
+        uint256 presentValue;
+        uint256 discountRate;
+        uint256 borrowAmount;
+        // calculate borrow amount
+        {
+            // fetch present value and discount rate from Oracle
+            (presentValue, discountRate) = IInstantNFTOracle(WITHDRAW_ORACLE)
+                .getPresentValueAndDiscountRate(
+                    collateralTokenId,
+                    collateralAmount,
+                    reserve.currentStableBorrowRate
+                );
+
+            uint256 presentValueInBorrowAsset = _calculatePresentValueInBorrowAsset(
+                    borrowAsset,
+                    presentValue
+                );
+            borrowAmount = presentValueInBorrowAsset.percentMul(
+                PercentageMath.PERCENTAGE_FACTOR - ps._loanCreationFeeRate
+            );
+        }
+
+        // validate borrow asset can be borrowed from lending pool
+        ValidationLogic.validateInstantWithdrawBorrow(
+            reserveCache,
+            borrowAsset,
+            borrowAmount
+        );
+
+        // handle asset
+        {
+            // transfer collateralAsset to reserveAddress
+            IERC1155(collateralAsset).safeTransferFrom(
+                msg.sender,
+                VAULT_CONTRACT,
+                collateralTokenId,
+                collateralAmount,
+                ""
+            );
+
+            // mint debt token for reserveAddress and transfer borrow asset to borrower
+            (
+                ,
+                reserveCache.nextTotalStableDebt,
+                reserveCache.nextAvgStableBorrowRate
+            ) = IStableDebtToken(reserveCache.stableDebtTokenAddress).mint(
+                VAULT_CONTRACT,
+                VAULT_CONTRACT,
+                borrowAmount,
+                discountRate
```

Comment:
why it's not reserve.currentStableBorrowRate?

Reply:
Current requirement is using discountRate as borrow rate.